module/programs/helper.js

```diff
@@ -494,21 +500,72 @@ module.exports = class ProgramsHelper {
         }
         queryData.data.startDate ={"$lte": new Date()}
         queryData.data.endDate ={"$gte": new Date()}
+        
         let targetedPrograms = await this.list(
           pageNo,
           pageSize,
           searchText,
           queryData.data,
-          ["name", "externalId","components"]
+          projection
         );
-             
-        if ( targetedPrograms.success && targetedPrograms.data && targetedPrograms.data.data.length > 0) {
+        
+        return resolve({
+          success: true,
+          message: constants.apiResponses.TARGETED_PROGRAMS_FETCHED,
+          data: targetedPrograms.data
+        });
+
+      } catch (error) {
+
+        return resolve({
+          success : false,
+          message : error.message,
+          data : {}
+        });
+
+      }
+
+    })
+  }
+
+  /**
+   * List of programs related to a user.
```

Comment:
@VISHNUDAS-tunerlabs As discussed please check if this function is really required